### DISCUSSION TITLE
Unable to Visualize Go Runtime Metrics in SkyWalking Under Service Instance Golang Section

### DISCUSSION THREAD

**Discussion by vishalkadam18** [apache/skywalking#12562] [Q&A]
Dear Team,

I successfully integrated the SkyWalking Go agent with my Go application, and the application is being monitored in the SkyWalking UI. However, I'm unable to see the following Go runtime metrics visualized under the service instance Golang section:

meter_instance_golang_memory_heap_labeled
meter_instance_golang_live_goroutines_num
meter_instance_golang_os_threads_num
meter_instance_golang_sched_latencies
meter_instance_golang_gc_count_labeled
meter_instance_golang_gc_pauses
meter_instance_golang_gc_heap_goal
meter_instance_golang_heap_frees
meter_instance_golang_heap_alloc_rate
meter_instance_golang_gc_heap_objects
meter_instance_golang_gc_heap_frees_by_size
meter_instance_golang_gc_heap_allocs_by_size
meter_instance_golang_cgo_calls
meter_instance_golang_metadata_mspan_labeled
meter_instance_golang_metadata_mcache_labeled
Is there a specific configuration or step required to enable the visualization of these metrics, or is this a known limitation?

Thank you!

**Comment by wu-sheng** [comment]
I don't think there is a config. @mrproliu ?

**Reply by rajeshnayak23** [reply]
How can we able to visualise these above metrics data under service instance golang section using go agent ?

**Reply by wu-sheng** [reply]
It should be automatically showing up in the UI. Do you check the data in the storage? Are there data existing?

**Reply by rajeshnayak23** [reply]
Yes , I have checked the storage but these data are not existing in the storage .  

How does it happens the other service data are visible in the ui but not the service instance golang data ?

**Reply by wu-sheng** [reply]
If it doesn't exist, the agent somehow doesn't report it.

https://skywalking.apache.org/docs/skywalking-go/latest/en/agent/support-plugins/#metrics-plugins

**Reply by mrproliu** [reply]
The metrics should be uploaded to the OAP automatically, for now, there is no way to disable it(only can change the default report interval). Could you help to provide the following information:
1. Which version of go agent?
2. Which version of the OAP server? 
3. Have you added the `go-runtime` in the config `agent-analyzer.default.meterAnalyzerActiveFiles` of OAP configuration?

**Reply by rajeshnayak23** [reply]
1.Go Agent Version: v1.5.0
2.OAP server Version: v9.7.0
3. yes it already exist  the go-runtime in the config agent-analyzer.default.meterAnalyzerActiveFiles of OAP configuration.

**Reply by mrproliu** [reply]
We don't have 1.5.0 version of go agent. Current is 0.4.0.
